core/src/main/java/org/springframework/security/oauth2/server/authorization/converter/OAuth2AuthorizationCodeAuthenticationTokenConverter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.converter;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames;
+import org.springframework.security.oauth2.server.authorization.authentication.OAuth2AuthorizationCodeAuthenticationToken;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Converts from a {@link HttpServletRequest} to an {@link OAuth2AuthorizationCodeAuthenticationToken} that can be authenticated. The
+ * converter does not validate the request but only performs a conversion.
+ * @author Madhu Bhat
+ */
+public class OAuth2AuthorizationCodeAuthenticationTokenConverter implements Converter<HttpServletRequest, OAuth2AuthorizationCodeAuthenticationToken> {
+
+	@Override
+	public OAuth2AuthorizationCodeAuthenticationToken convert(HttpServletRequest request) {
+		return new OAuth2AuthorizationCodeAuthenticationToken(
+				request.getParameter(OAuth2ParameterNames.CODE),
+				request.getParameter(OAuth2ParameterNames.CLIENT_ID),
```

Comment:
Please see my earlier [comment](https://github.com/spring-projects-experimental/spring-authorization-server/pull/79#issuecomment-625897548):

> so the expectation is that `OAuth2ClientAuthenticationToken` #72 is available in the `SecurityContext` - no need to validate client_id

The other constructor should be called - the one accepting `Authentication clientPrincipal`